### PR TITLE
allows vox to change their hair colors

### PR DESCRIPTION
**Pull request by robintime**
IEATTHATLUNCH
hair1

before someone asks why i wouldnt just use a color overlay on one greyscale sprite: the colors of vox skin tones are not darker/lighter variations of a single color, and i wanted the hair colors to be exact to palettes of the skin colors. this is best seen on the grey  skin tone:
hair3
you can see here that 1 is a desaturated green, while 2 is a comparatively saturated blue. this isn't replicable on a greyscale sprite with a color overlay.

for consistency, the different sprites were made by swapping colors. here is the palette i ended up with:
hair2
(i forgot to save light green sorry)

:cl:
 * rscadd: adds an input menu that allows the player to select a hair color from a pre-defined list of seven colors, corresponding to the six vox skin tones and the current default hair color
 * tweak: changes clockwork's vox hairstyles to use the same palette as the older hairs
 * spellcheck: removes erroneous spaces in some places in the code (it bothered me)

### CODE REVIEW

**Comment by SonixApache** [comment]
added not adds
fix your fukken commit messages

**Comment by robintime** [comment]
> added not adds fix your fukken commit messages

im about to have added my fist to your face

**Comment by SonixApache** [comment]
that can wait till after you fix your PR

**Comment by Kromkar** [comment]
i hate vox but this seems nice

**Comment by Kurfursten** [comment]
Not compatible with hair dye?

**Comment by robintime** [comment]
> Not compatible with hair dye?

youre evil

**Comment by robintime** [comment]
im actually not sure how to implement the hair dye for this, because these hair colors are static sprites and not rgb values; i could make it give you a selection of vox hair colors if your species is vox when you use it in your hand, but that eliminates the ability for vox to make it hot pink and then recolor someone's hair without their consent - it would be a sorta tacky solution

what i could see working is some bit of code that detects "oh this color is close to emerald" and then makes the v_hair variable emerald when used on vox, but i have no idea how to write that (something that sorts color values into categories based on their proximity to certain colors, i mean)

**Comment by Kurfursten** [comment]
Here's what you do, it's simple. Assign all 7 colors a set of RGB values. You can either make these datumized or use an associative list.

When the color is applied to a vox, run through all 7 sets and compare the RGB values with a simple subtraction and pick the one with the least difference. For example...

Pink vs colors
image
Pink 255 174 201
Testing color 1 / brown: 162 107 56
- Individual differences 93 67 150
- Total diff 310
- 310 is less than ARBITRARILY_LARGE_NUMBER, now using brown


Testing color 2 / green brown: 147 126 61
- Individual differences 108 48 140
- Total diff 296
- 296 is less than 310, now using green brown

Continue to loop through all colors until you have the best one

**Comment by ShiftyRail** [comment]
The technicolor furry metaclub grows even more powerful by the day.

**Comment by Theophrastus1** [comment]
vox erp rgb metaclub
birds aren't real

**Comment by Armadingus** [comment]
Hello merge when??????

**Comment by robintime** [comment]
> Hello merge when??????

merge when i make it compatible with hair dye

**Comment by Kromkar** [comment]
> > Hello merge when??????
> 
> merge when i make it compatible with hair dye

Who gives a fuck about hair dye?

Merge and then do hair dye later

File: code/game/objects/items/weapons/cosmetics.dm [hunk 1]
```diff
@@ -264,14 +264,59 @@
...
+
+		voxhaircolorlist["brown"] = list(162,107,56)
+		voxhaircolorlist["greenbrown"] = list(147, 126, 61)
+		voxhaircolorlist["lightgreen"] = list(132, 138, 64)
+		voxhaircolorlist["azure"] = list(112, 126, 93)
+		voxhaircolorlist["green"] = list(87, 123, 119)
+		voxhaircolorlist["emerald"] = list(65, 136, 98) //299
+		voxhaircolorlist["grey"] = list(192, 192, 192) //876
+		
+		to_chat(world,"vhcl: [voxhaircolorlist] - [voxhaircolorlist.len]")
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/cosmetics.dm]: remove all your debug code and I will merge this

**Pull request merged**